src/rust/engine/process_execution/src/local.rs

```diff
@@ -850,7 +850,11 @@ pub fn setup_run_sh_script(
     let formatted_assignment = format!("{key}={arg_str}");
     env_var_strings.push(formatted_assignment);
   }
-  let stringified_env_vars: String = env_var_strings.join(" ");
+  let stringified_env_var_export: String = if env_var_strings.is_empty() {
```

Comment:
You shouldn't need the type annotation here.

Reply:
I figured if it was needed before it would be now, but will remove if it works.